File: Swift/tests/TRNCHolidayValidatorTests.swift
import XCTest
@testable import YourModuleName

class TRNCHolidayValidatorTests: XCTestCase {
  var validator: TRNCHolidayValidator!

  override func setUp() {
    super.setUp()
    validator = TRNCHolidayValidator()
  }

  func testTRNCFixedHolidays() {
    XCTAssertTrue(validator.isHoliday(date: DateComponents(year: 2024, month: 8, day: 1).date!))
  }

  func testTRNCIslamicHolidays() {
    XCTAssertTrue(validator.isHoliday(date: DateComponents(year: 2024, month: 9, day: 15).date!))
  }
}

File: C#/tests/TRNCHolidayValidatorTests.cs
using System;
using Xunit;

namespace HolidayChecker.Tests
{
    public class TRNCHolidayValidatorTests
    {
        private TRNCHolidayValidator validator;

        public TRNCHolidayValidatorTests()
        {
            validator = new TRNCHolidayValidator(false);
        }

        [Fact]
        public void TestTRNCFixedHolidays()
        {
            Assert.True(validator.IsHoliday(new DateTime(2024, 8, 1)));
        }

        [Fact]
        public void TestTRNCIslamicHolidays()
        {
            Assert.True(validator.IsHoliday(new DateTime(2024, 9, 15)));
        }
    }
}
